tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -989,3 +997,530 @@ def progress_check():
+
+            # Sent indices as values
+            p = kclient.get_producer()
+            sent_count = 0
+            for idx in range(1, message_count + 1):
+                # Async message sending func
+                p.produce(topic,
+                          key=f"key_{idx}",
+                          value=f"{idx:04}",
+                          callback=acked)
+                # Make sure message sent, aka sync
+                p.flush()
+                sent_count += 1
+            # Return stats and errors
+            return sent_count, errors
+
+        def check_consecutive(numbers_list):
+            n = len(numbers_list) - 1
+            # Calculate iterative difference for the array. It should be 1.
+            return (sum(numpy.diff(sorted(numbers_list)) == 1) >= n)
+
+        # Prepare librdkafka python client
+        kclient = PythonLibrdkafka(self.redpanda)
+        # Messages to produce
+        message_count = 100
+        self.logger.info(
+            f"Producing {message_count} messages to {num_topics} random topics"
+        )
+        # Pick random topics to send messages to
+        total_topics = len(topic_names) - 1
+        if total_topics > num_topics:
+            random_topic_indices = [
+                random.randint(0, total_topics) for i in range(num_topics)
+            ]
+            next_topic_batch = []
+            while len(random_topic_indices) > 0:
+                next_topic_batch.append(
+                    topic_names[random_topic_indices.pop()])
+            new_topic_names = list(set(topic_names) - set(next_topic_batch))
+        else:
+            next_topic_batch = topic_names
+            new_topic_names = []
+
+        # Send messages
+        messages_sent = 0
+        errors = []
+        with concurrent.futures.ThreadPoolExecutor(16) as executor:
+            for c, thread_errors in executor.map(_send_messages,
+                                                 next_topic_batch):
+                messages_sent += c
+                errors += thread_errors
+        self.logger.info(f"Total of {messages_sent} messages sent")
+
+        total_errors = len(errors)
+        if total_errors > 0:
+            self.logger.error(f"{total_errors} Errors detected "
+                              "while sending messages")
+
+        # Select random topic from the list
+        target_topic = next_topic_batch[random.randint(0,
+                                                       len(next_topic_batch))]
+        # Consumer specific config
+        consumer_extra_config = {
+            "auto.offset.reset": "smallest",
+            "group.id": "topic_swarm_group"
+        }
+        self.logger.info(f"Start consuming from {target_topic}")
+        # Consumer
+        start_time_s = time.time()
+        consumer = kclient.get_consumer(consumer_extra_config)
+        numbers = []
+        # Message consuming loop
+        try:
+            consumer.subscribe([target_topic])
```

Comment:
nit: it's possible to subscribe to all topics within a single consumer group. It may be nice to do that here and ensure it's possible to consume all numbers from all the topics we produced to. Or maybe creating multiple consumers within a single consumer group that has all the topics we produced to. It'll likely complicate the validation logic though.

Reply:
Goal was to make this as simple as possible and that the runtime would be under 10 min.